include/mbedtls/ssl.h

```diff
@@ -382,6 +388,28 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
+static inline int mbedtls_ssl_chk_buf( uint8_t *cur, const uint8_t *end, size_t need )
+{
+    return( end < cur || (size_t)( end - cur ) < (size_t) need );
+}
+
+#define MBEDTLS_SSL_CHK_BUF_PTR( cur, end, need )			\
+    do {								\
+        if( mbedtls_ssl_chk_buf( ( cur ), ( end ), ( need ) ) != 0 )	\
+        {								\
+            return( MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );			\
+        }								\
+    } while( 0 )
```

Comment:
I disagree with this. Just because you want to avoid two print statements you are making the code over-complicated. I would suggest just having one version of the macro that always prints the error message BUT decrease the priority of the debug print to something higher than one. Also, in my opinion if the debug print is not available in some cases, I would just prefer to remove the print altogether. In any case, in my opinion the error code `MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL` is self explanatory.